File: find_user_by_email.py
import psycopg2
import os

# Get database URL from environment variable
DATABASE_URL = os.environ.get("DATABASE_URL") or os.environ.get("POSTGRES_URL")

if not DATABASE_URL:
    raise Exception("DATABASE_URL or POSTGRES_URL environment variable not set.")

def find_user_by_email(email):
    try:
        conn = psycopg2.connect(DATABASE_URL)
        cur = conn.cursor()
        cur.execute('SELECT * FROM users WHERE email = %s', (email,))
        row = cur.fetchone()
        if row:
            print('User record:', row)
        else:
            print('User not found.')
        cur.close()
        conn.close()
    except Exception as e:
        print('Error querying user:', e)

if __name__ == "__main__":
    find_user_by_email('developer@lotteryanalytics.app')

File: find_user_by_email.ts
const pool = require("./app/src/lib/db.ts");

async function findUserByEmail(email) {
  try {
    const res = await pool.query('SELECT * FROM users WHERE email = $1', [email]);
    if (res.rows.length === 0) {
      console.log('User not found.');
    } else {
      console.log('User record:', res.rows[0]);
    }
  } catch (err) {
    console.error('Error querying user:', err);
  } finally {
    await pool.end();
  }
}

findUserByEmail('developer@lotteryanalytics.app');
